source/Calamari.Common/Features/Scripting/Bash/BashScriptBootstrapper.cs

```diff
@@ -55,6 +59,15 @@ public static string PrepareConfigurationFile(string workingDirectory, IVariable
             return configurationFile;
         }
 
+        static IEnumerable<string> GetVariableNameAndValueDeclaration(IEnumerable<EncryptedVariable> variables)
+        {
+            return variables.Select(variable =>
+                                    {
+                                        var variableValue = $@"$(get_octopusvariable ""{variable.Name.Replace("\"", "\\\"")}"")";
```

Comment:
So far the only case I've found that I have to manually add delimiters to is `"`. This is because the final generated command has the format:

```
 declare -grA octopus_parameters=(["Octopus.Agent.ProgramDirectoryPath"]="$(get_octopusvariable "Octopus.Agent.ProgramDirectoryPath")" ["Octopus.Sashimi.ScriptFunctions.Registration"]="$(get_octopusvariable "Octopus.Sashimi.ScriptFunctions.Registration")" ["Octopus.Deployment.Id"]="$(get_octopusvariable "Octopus.Deployment.Id")" ["Octopus.Deployment.Name"]="$(get_octopusvariable "Octopus.Deployment.Name")" ["Octopus.Deployment.Created"]="$(get_octopusvariable "Octopus.Deployment.Created")" ["Octopus.Deployment.CreatedUtc"]="$(get_octopusvariable "Octopus.Deployment.CreatedUtc")"
```

Reply:
backtick characters also need to be escaped.

<img width="280" alt="image" src="https://github.com/user-attachments/assets/275b54e5-5d6b-4205-9150-24f7455de43a" />

<img width="493" alt="image" src="https://github.com/user-attachments/assets/609f4f08-8295-4b89-812c-c50677ee4f89" />

Reply:
I ended up raising an issue 
https://github.com/OctopusDeploy/Issues/issues/9255